### PR TITLE
Adding Package for wxMPL

### PR DESCRIPTION
**Pull request by ChrisBarker-NOAA**
Pure Python -- should be straightforward.

@conda-forge/staged-recipes

Once the PR is ready for review, please mention one of the teams above in a
new comment. i.e. `@conda-forge/help-some-language, ready for review!`
Then, a bot will label the PR as 'review-requested'.

[1]: https://conda-forge.org/docs/maintainer/infrastructure.html#conda-forge-admin-please-ping-team
[2]: https://gitter.im/conda-forge/conda-forge.github.io

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by ChrisBarker-NOAA** [comment]
@conda-forge/help-python, ready for review!

Linux_64 is failing:

`ImportError: libXxf86vm.so.1: cannot open shared object file: No such file or directory`

That's probably because this is using wxPython which cannot be imported unless there is a running Xserver. It won't even import on a headless server.

Is there a  test image with an X server running? Or can we just ignore this failure?

**Comment by ocefpaf** [comment]
You can add the CDTs to satisfy that dependency: https://conda-forge.org/docs/maintainer/knowledge_base.html#libgl

File: recipes/wxmpl/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
+{% set name = "wxmpl" %}
+{% set version = "2.3.0" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  fn: wxmpl-v{{ version }}.tar.gz
```
Comment (ocefpaf) [inline recipes/wxmpl/meta.yaml:9]:
```suggestion
```

File: recipes/wxmpl/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  url: https://github.com/NOAA-ORR-ERD/wxmpl/archive/v{{ version }}.tar.gz
+  sha256: 20258f22b0affbd56968af8b0743e99bf8ade4845cbb4ce057398cd693787c7e
+
+build:
+  number: 0
+  noarch: python
+#  script: pip install --no-deps --no-cache-dir ./
```
Comment (ocefpaf) [inline recipes/wxmpl/meta.yaml:16]:
```suggestion
```

File: recipes/wxmpl/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  license_family: BSD
+  license_file: LICENSE.txt
+  dev_url: https://github.com/NOAA-ORR-ERD/wxmpl
+  summary: 'Painless matplotlib embedding in wxPython'
+
+extra:
+  recipe-maintainers:
+    # GitHub IDs for maintainers of the recipe.
+    # Always check with the people listed below if they are OK becoming maintainers of the recipe. (There will be spam!)
+    - ChrisBarker-NOAA
```
Comment (ocefpaf) [inline recipes/wxmpl/meta.yaml:47]:
```suggestion
    - ChrisBarker-NOAA
```

File: recipes/wxmpl/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+build:
+  number: 0
+  noarch: python
+#  script: pip install --no-deps --no-cache-dir ./
+  script: {{ PYTHON }} -m pip install . -vv
+
+requirements:
+  host:
+    - python <=3.10
```
Comment (ocefpaf) [inline recipes/wxmpl/meta.yaml:21]: Readme says `>=3.8`, setup.py metadata says `=2.7`.

**Comment by ChrisBarker-NOAA** [comment]
Note that:
>  https://conda-forge.org/docs/maintainer/knowledge_base.html#libgl

talks about libgl -- in this case, I only need X, not OpenGL -- it can't be that rare to have an X based package that isn't using GL. (though nice to see this, as I also need to get a pyopengl-accelerate working). It would be good to add a section to the docs for X without GL.

(I'll go post a  PR for that)

In this case, as a pure-python package, I don't need the build deps, so I added the X libs to yum_requirements -- borrowed from the wxPython recipe.

--we'll see if that works.

In the long run, it would be nice to have a simpler configuration option for "needs X" or "needs GL", and some of us have no clue which individual libraries are needed.

File: recipes/wxmpl/meta.yaml (hunk 4 above)
Comment (ChrisBarker-NOAA) [reply recipes/wxmpl/meta.yaml]: Honestly, I haven't tested it on very many versions lately -- but I'll get it all in sync.

**Comment by ChrisBarker-NOAA** [comment]
I think it's finally all cleaned up now.

@help-python, ready for review!

**Comment by ChrisBarker-NOAA** [comment]
Thanks!

**Pull request merged**